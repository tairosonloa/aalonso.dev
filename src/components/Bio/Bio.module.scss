@import 'src/styles/utils.scss';
@import 'src/styles/palette.scss';

.container {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding: grid_units(17) grid_units(4);
  background-color: $grey-white;
  text-align: center;
}

.header {
  padding-bottom: grid_units(8);
}

.image {
  width: grid_units(34);
  height: grid_units(34);
  padding: grid_units(4);
  border-radius: 50%;
}

.socialMedia {
  margin-top: grid_units(3);
  margin-bottom: grid_units(6);
  list-style: none;

  & > li {
    display: inline-block;
    margin: 0 grid_units(2);

    & > a {
      text-decoration: none;

      & > *:hover {
        color: $burgundy;
      }
    }
  }
}

.bio {
  max-width: grid_units(160);
  padding-right: grid_units(4);
  padding-left: grid_units(4);
  text-align: center;
}
