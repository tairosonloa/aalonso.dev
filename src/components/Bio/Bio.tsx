import React, { FC } from 'react'
import { SiteMetadata } from '../../../site.metadata'
import { Icon, IconName } from '../Icon/Icon'
import { Text, TextType } from '../Text/Text'
import styles from './Bio.module.scss'

export const Bio: FC = () => {
  const { socialMediaUrls, employerInfo } = SiteMetadata

  return (
    <div className={styles.container}>
      <div className={styles.header}>
        <Text textType={TextType.TITLE}>About me</Text>
        <Text textType={TextType.TITLE_SMALL}>Let me introduce myself first.</Text>
      </div>
      <img src="/images/me.jpeg" alt="" className={styles.image} />
      <ul className={styles.socialMedia}>
        <li>
          <a
            href={socialMediaUrls.github}
            target="_blank"
            rel="noopener noreferrer"
            data-testid="github-link"
          >
            <Icon name={IconName.GITHUB} data-testid="github" />
          </a>
        </li>
        <li>
          <a
            href={socialMediaUrls.linkedin}
            target="_blank"
            rel="noopener noreferrer"
            data-testid="linkedin-link"
          >
            <Icon name={IconName.LINKEDIN} data-testid="linkedin" />
          </a>
        </li>
        <li>
          <a
            href={`mailto:${socialMediaUrls.emailAddress}`}
            target="_blank"
            rel="noopener noreferrer"
            data-testid="email-link"
          >
            <Icon name={IconName.EMAIL} data-testid="email" />
          </a>
        </li>
      </ul>
      <Text className={styles.bio}>
        I&apos;m a full stack developer full-time working for{' '}
        <a href={employerInfo.website} target="_blank" rel="noopener noreferrer">
          {employerInfo.name}
        </a>
        , while studying for a Master&apos;s Degree in Computer Science and Engineering on weekends.
        Focused on TypeScript with NodeJS and React, I use these technologies both professionally
        and for personal side projects, like this website.
      </Text>
      <Text className={styles.bio}>
        I&apos;ve always been proactive, enthusiastic and curious, so I never stop learning and
        doing new stuff. I&apos;ve been playing with python for artificial intelligence and machine
        learning for a few years, specially around neural networks. Free software lover and DevOps
        soul, I&apos; consider myself an experienced Linux system administrator, so I self
        maintained my own virtual private server for years.
      </Text>
    </div>
  )
}
