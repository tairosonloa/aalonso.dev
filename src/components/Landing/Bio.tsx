import React, { FC } from 'react'
import Emoji from '../Emoji'

const Bio: FC = () => {
  return (
    <div className="flex flex-col space-y-12 divide-y divide-gray-200 mb-24">
      <div>
        <h1 className="text-4xl md:text-5xl lg:text-6xl font-extrabold leading-tight tracking-tight text-gray-100  sm:leading-10   md:leading-14">
          Hi, I&apos;m Aitor <Emoji emoji="👋" label="Waving hand" />
        </h1>
        <p className="mt-3 mb-6 md:mb-7 text-gray-300 md:w-full text-base md:text-lg font-medium">
          I&apos;m a full-stack software engineer <Emoji emoji="👨🏻‍💻" label="Man with a computer" />{' '}
          who uses Domain Driven Design and Clean Architectures to build fast{' '}
          <Emoji emoji="⚡" label="Lightning" />, efficient and reliable back-end APIs.
        </p>
      </div>
    </div>
  )
}

export default Bio
