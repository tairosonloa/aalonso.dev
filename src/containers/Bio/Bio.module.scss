@import 'src/styles/utils.scss';
@import 'src/styles/palette.scss';

.container {
  margin: auto;
  min-height: 100vh;
  max-width: grid_units(200);
  display: flex;
  flex-direction: column-reverse;
  justify-content: center;
  align-items: center;

  & > div {
    margin-top: grid_units(8);
    margin-right: grid_units(5);
    margin-left: grid_units(5);
  }

  @include screenIn(desktop) {
    flex-direction: row;
    align-items: center;
    justify-content: space-between;
  }
}

.imageAndLinks {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.image {
  width: grid_units(34);
  height: grid_units(34);
  
  img {
    border-radius: 50%;
  }
}

.socialMediaLinks {
  margin-top: 2rem;
  margin-bottom: 2rem;

  i {
    color: $light-gray;
  }
}

.bio {
  max-width: grid_units(130);
  text-align: center;

  @include screenIn(desktop) {
    text-align: left;
  }
}