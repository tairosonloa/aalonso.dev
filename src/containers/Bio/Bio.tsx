import React, { FC } from 'react'
import { SiteMetadata } from '../../../site.metadata'
import { SectionContainer } from '../../components/DataDisplays/SectionContainer/SectionContainer'
import { SectionHeader } from '../../components/DataDisplays/SectionHeader/SectionHeader'
import { Typography } from '../../components/DataDisplays/Typography/Typography'
import { SocialMediaLinks } from '../../components/SocialMediaLinks/SocialMediaLinks'
import styles from './Bio.module.scss'

export const Bio: FC = () => {
  const { employerInfo } = SiteMetadata

  return (
    <SectionContainer darkBackground>
      <SectionHeader title="About me" subtitle="Let me introduce myself first." />
      <img src="/images/me.jpeg" alt="" className={styles.image} />
      <SocialMediaLinks className={styles.socialMediaLinks} />
      <Typography className={styles.bio}>
        I&apos;m a full stack developer full-time working for{' '}
        <a href={employerInfo.website} target="_blank" rel="noopener noreferrer">
          {employerInfo.name}
        </a>
        , while studying for a Master&apos;s Degree in Computer Science and Engineering on weekends.
        Focused on TypeScript with NodeJS and React, I use these technologies both professionally
        and for personal side projects, like this website.
      </Typography>
      <Typography className={styles.bio}>
        I&apos;ve always been proactive, enthusiastic and curious, so I never stop learning and
        doing new stuff. I&apos;ve been playing with python for artificial intelligence and machine
        learning for a few years, specially around neural networks. Free software lover and DevOps
        soul, I&apos; consider myself an experienced Linux system administrator, so I self
        maintained my own virtual private server for years.
      </Typography>
    </SectionContainer>
  )
}
