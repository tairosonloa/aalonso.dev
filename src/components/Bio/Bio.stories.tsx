import React from 'react'
import { dimensionsDecorator } from '../../../.storybook/utils/Dimensions'
import { Bio } from './Bio'

export default {
  title: 'Common|Bio',
  component: Bio,
  decorators: [dimensionsDecorator()],
}

export const Default = (): JSX.Element => <Bio />
