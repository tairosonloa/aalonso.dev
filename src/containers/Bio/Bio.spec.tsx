import { render } from '@testing-library/react'
import * as React from 'react'
import { Bio } from './Bio'

describe('<Bio/>', () => {
  it('should display the default message', () => {
    const title = 'About me'
    const subtitle = 'Let me introduce myself first.'

    const renderResult = render(<Bio />)

    expect(renderResult.getByText(title)).toBeTruthy()
    expect(renderResult.getByText(subtitle)).toBeTruthy()
  })

  it('should display the image', () => {
    const imagePath = '/images/me.jpeg'

    const renderResult = render(<Bio />)
    expect(renderResult.getByAltText('')).toHaveAttribute('src', imagePath)
  })

  it('should match the snapshot', () => {
    const renderResult = render(<Bio />)
    expect(renderResult).toMatchSnapshot()
  })
})
