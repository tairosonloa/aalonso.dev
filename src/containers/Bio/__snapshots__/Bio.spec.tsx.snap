// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Bio/> should match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container dark"
        id="bio"
      >
        <div
          class="container hidden"
        >
          <h2
            class="common title"
            data-testid="section-title"
          >
            About me
          </h2>
          <h3
            class="common titleSmall"
            data-testid="section-subtitle"
          >
            Let me introduce myself first.
          </h3>
        </div>
        <div
          class="hidden"
        >
          <img
            alt=""
            class="image"
            src="/images/me.jpeg"
          />
          <ul
            class="container socialMediaLinks"
          >
            <li>
              <a
                data-testid="github-link"
                href="https://github.com/tairosonloa"
                rel="noopener noreferrer"
                target="_blank"
              >
                <i
                  class="ri-github-fill icon"
                  data-testid="github"
                />
                <span
                  class="hiddenTextForAccessibility"
                >
                  github
                </span>
              </a>
            </li>
            <li>
              <a
                data-testid="linkedin-link"
                href="https://www.linkedin.com/in/aitoralonsonunez/"
                rel="noopener noreferrer"
                target="_blank"
              >
                <i
                  class="ri-linkedin-box-fill icon"
                  data-testid="linkedin"
                />
                <span
                  class="hiddenTextForAccessibility"
                >
                  linkedin
                </span>
              </a>
            </li>
            <li>
              <a
                data-testid="email-link"
                href="mailto:aitor@aalonso.dev"
                rel="noopener noreferrer"
                target="_blank"
              >
                <i
                  class="ri-mail-fill icon"
                  data-testid="email"
                />
                <span
                  class="hiddenTextForAccessibility"
                >
                  email
                </span>
              </a>
            </li>
          </ul>
          <p
            class="common paragraph bio"
          >
            I'm a full stack developer full-time working for
             
            <a
              href="https://acidtango.com/"
              rel="noopener noreferrer"
              target="_blank"
            >
              Acid Tango
            </a>
            , while studying for a Master's Degree in Computer Science and Engineering on weekends. Focused on TypeScript with Node.js and React, I use these technologies both professionally and for personal side projects, like this website.
          </p>
          <p
            class="common paragraph bio"
          >
            I'm proactive, enthusiastic and curious, so I never stop learning and doing new stuff. I've been playing with python for artificial intelligence and machine learning for a few years, specially around neural networks. Free software lover and DevOps soul, I'll consider myself an experienced Linux system administrator, so I self maintained my own virtual private server for years.
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container dark"
      id="bio"
    >
      <div
        class="container hidden"
      >
        <h2
          class="common title"
          data-testid="section-title"
        >
          About me
        </h2>
        <h3
          class="common titleSmall"
          data-testid="section-subtitle"
        >
          Let me introduce myself first.
        </h3>
      </div>
      <div
        class="hidden"
      >
        <img
          alt=""
          class="image"
          src="/images/me.jpeg"
        />
        <ul
          class="container socialMediaLinks"
        >
          <li>
            <a
              data-testid="github-link"
              href="https://github.com/tairosonloa"
              rel="noopener noreferrer"
              target="_blank"
            >
              <i
                class="ri-github-fill icon"
                data-testid="github"
              />
              <span
                class="hiddenTextForAccessibility"
              >
                github
              </span>
            </a>
          </li>
          <li>
            <a
              data-testid="linkedin-link"
              href="https://www.linkedin.com/in/aitoralonsonunez/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <i
                class="ri-linkedin-box-fill icon"
                data-testid="linkedin"
              />
              <span
                class="hiddenTextForAccessibility"
              >
                linkedin
              </span>
            </a>
          </li>
          <li>
            <a
              data-testid="email-link"
              href="mailto:aitor@aalonso.dev"
              rel="noopener noreferrer"
              target="_blank"
            >
              <i
                class="ri-mail-fill icon"
                data-testid="email"
              />
              <span
                class="hiddenTextForAccessibility"
              >
                email
              </span>
            </a>
          </li>
        </ul>
        <p
          class="common paragraph bio"
        >
          I'm a full stack developer full-time working for
           
          <a
            href="https://acidtango.com/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Acid Tango
          </a>
          , while studying for a Master's Degree in Computer Science and Engineering on weekends. Focused on TypeScript with Node.js and React, I use these technologies both professionally and for personal side projects, like this website.
        </p>
        <p
          class="common paragraph bio"
        >
          I'm proactive, enthusiastic and curious, so I never stop learning and doing new stuff. I've been playing with python for artificial intelligence and machine learning for a few years, specially around neural networks. Free software lover and DevOps soul, I'll consider myself an experienced Linux system administrator, so I self maintained my own virtual private server for years.
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
